### PR TITLE
reintroduce octokit retry and throttling plugins

### PR DESCRIPTION
**Pull request by softprops**
With the last stream up updates came an upgrade to the `@actions/github` package which made it difficult to configure plugins as one normally would with octokit. 

This had the effect of disabling automatic retry behavior with now has become a feature request. 

It would seem the community around these actions octokit integrations with actions has moving around and there's a suggestion [here](https://github.com/actions/toolkit/issues/334) to avoid using `@actions/github` and use `@octokit/action` instead. I don't have a personal preference so long as the result ends up playing nicer with the octokit ecosystem

### CODE REVIEW

**Comment by softprops** [comment]
Im now seeing ncc generate 3 additional files. I need to investigate why before merging

```
  3kB  dist/fixasync.js
  17kB  dist/sandbox.js
  37kB  dist/contextify.js
```

File: dist/contextify.js [hunk 1]
```diff
@@ -0,0 +1,1139 @@
+/* global host */
+/* eslint-disable block-spacing, no-multi-spaces, brace-style, no-array-constructor, new-cap, no-use-before-define */
+
+'use strict';
```
Comment (softprops) [inline dist/contextify.js:4]: ncc is now generating these files and I don't know why

**Comment by BotellaA** [comment]
@softprops any update on this PR? It would be great to have this retry feature!! :pray:

File: src/main.ts [hunk 2]
```diff
@@ -32,13 +32,9 @@ async function run() {
...
 
-    // const oktokit = GitHub.plugin(
-    //   require("@octokit/plugin-throttling"),
-    //   require("@octokit/plugin-retry")
-    // );
-
-    const gh = getOctokit(config.github_token, {
-      //new oktokit(
+    const OctokitWithPlugins = Octokit.plugin(retry, throttling);
+    const gh = new OctokitWithPlugins({
```
Comment (Puneeth-n) [inline src/main.ts:36]: This is so cool! I didn't know! :)

**Pull request closed without merging**